config/configuration.rb

```diff
@@ -1,23 +1,115 @@
+require 'pathname'
+require 'dotenv'
+
 # dashboard app specific configuration
 class Configuration
-  class << self
-    attr_writer :app_development_enabled
-    attr_writer :app_sharing_enabled
-
-    # FIXME: temporary
-    attr_accessor :app_sharing_facls_enabled
-    alias_method :app_sharing_facls_enabled?, :app_sharing_facls_enabled
-
-    def app_development_enabled?
-      return @app_development_enabled if defined? @app_development_enabled
-      ENV['OOD_APP_DEVELOPMENT'].present? || DevRouter.base_path.exist?
-    end
-    alias_method :app_development_enabled, :app_development_enabled?
-
-    def app_sharing_enabled?
-      return @app_sharing_enabled if defined? @app_sharing_enabled
-      @app_sharing_enabled = ENV['OOD_APP_SHARING'].present?
-    end
-    alias_method :app_sharing_enabled, :app_sharing_enabled?
+  attr_writer :app_development_enabled
```

Comment:
This setter *may* not be necessary since we allow this to be set through environment variables (may require a little elbow grease to make env vars that call bash code to check if usr app path exists if a site wants that level of customization).

Also I worry that we allow configuration options to be mutated later in the code, but we use these configuration options when building the app. Maybe we make `Configuration` objects immutable?

Reply:
Discussed offline. Thats a good concern. The original code provided a solution for this in that config_root and configuration where two separate objects, and only config_root was usable in bin/setup-production. I'll open a separate issue in myjobs on possibly adding some protection to ensure only immutable configuration options are used in the bin/setup-production script.